vignettes/epidist.Rmd

```diff
@@ -1,201 +1,250 @@
+We suppose that a sample of individuals of size `sample_size` are observed:
+
+```{r}
+sample_size <- 200
 ```
 
-Plot the mean difference between continuous and discrete event time:
+This sample size corresponds to `r 100 * round(sample_size / nrow(obs_cens_trunc), 3)`% of the data.
 
-```{r censor_delay}
-censor_delay <- calculate_censor_delay(truncated_obs)
-plot_censor_delay(censor_delay)
+```{r}
+obs_cens_trunc_samp <-
+  obs_cens_trunc[sample(seq_len(.N), sample_size, replace = FALSE)]
 ```
 
-### Model
+With our censored, truncated, and sampled data, we are now ready to try to recover the underlying delay distribution using `epidist`.
 
-Adjust for right truncation and date censoring using a latent variable approach.
+# Fit the model {#fit}
 
-```{r}
-latent_truncation_censoring_fit <- latent_truncation_censoring_adjusted_delay(
-  data = truncated_obs, cores = 4, refresh = 0
-)
+If we had access to the data `obs`, then it would be simple to estimate the delay distribution (Figure \@ref(fig:obs-est)).
+
+(ref:obs-est) The histogram of delays matches closely the underlying distribution. Edit: at the moment something is wrong here.
+
+```{r obs-est, fig.cap="(ref:obs-est)"}
+ggplot() +
+  geom_histogram(data = obs, aes(x = delay, y = stat(count / sum(count)))) +
+  geom_function(
+    data = data.frame(x = c(0, 30)),
+    aes(x = x),
+    fun = dlnorm,
+    meanlog = secondary_dist[["meanlog"]],
+    sdlog = secondary_dist[["sdlog"]],
+  ) +
+  labs(x = "Delay between primary and secondary event (days)", y = "Probability density") +
+  theme_minimal()
 ```
 
-### Summarise model posterior and compare to known truth
+However, with only access to `obs_cens_trunc_samp`, we must use a statistical model to avoid severely biased estimates!
 
-Combine models into a named list.
+The main `epidist` model function is `latent_truncation_censoring_adjusted_delay`.
+<!-- In a future vignette, we will explain in more detail the structure of the model. -->
+The parameters of the model are inferred using Bayesian inference.
+In particular, we use the the No-U-Turn Sampler (NUTS) Markov chain Monte Carlo (MCMC) algorithm via the [`brms`](https://paul-buerkner.github.io/brms/) R package [@brms].
 
 ```{r}
-models <- list(
-  "Latent variable truncation and censoring adjusted" = latent_truncation_censoring_fit
+fit <- latent_truncation_censoring_adjusted_delay(
+  data = obs_cens_trunc_samp, cores = 4, refresh = 0
 )
 ```
 
-Extract and summarise lognormal posterior estimates.
+The `fit` object is a `brmsfit` object containing MCMC samples from each of the parameters (Table \@ref(tab:pars)) in the model.
+Many of these are so-called "latent variables".
 
-```{r lognormal-draws}
-draws <- models |>
-  map(extract_lognormal_draws) |>
-  rbindlist(idcol = "model") |>
-  rbind(epinowcast_draws, use.names = TRUE)
+```{r pars}
+pars <- fit$fit@par_dims |>
+  map(.f = function(x) if(identical(x, integer(0))) return(1) else return(x))
```

Comment:
There are other choices that provide more functionality but are heavier like `tidybayes` (heavier in the sense of them themselves having more dependencies).

Reply:
I just didn't know an easier way to do this. Open to moving to `brms` / `posterior` / `tidybayes`. This is probably a one off thing for the vignette rather than a function in the package so not sure how much it matters.

Reply:
We already have `posterior` in `Imports:`

Reply:
I think using poosterior might be a good idea. What you have right now is a bit scary

Reply:
Will try doing. I don't know the function in `posterior` to do this off the top of my head or if there is one.

Reply:
I didn't find an easy way to do this with `posterior`.